mcs/class/corlib/System/Delegate.cs

```diff
@@ -495,8 +495,11 @@ public override bool Equals (object obj)
 
 		public override int GetHashCode ()
 		{
-			/* same implementation as CoreCLR */
-			return GetType ().GetHashCode ();
+			MethodInfo m;
+
+			m = Method;
+
+			return (m != null ? m.GetHashCode () : GetType ().GetHashCode ()) ^ (m_target != null ? m_target.GetHashCode () : 0);
```

Comment:
You could write that as `(Method?.GetHashCode() ?? GetType ().GetHashCode ())` to reduce code length.